### PR TITLE
OUT-479 Task card overflow issue

### PR DESCRIPTION
**Pull request by arpandhakal**
- also introduced divider in CU view

### Tasks

- [Task card overflow issue](https://linear.app/copilotplatforms/issue/OUT-479/task-card-overflow-issue)

### Changes

- [x] Fixed task card overflow in board view by truncating text which are more than 22 char for assignee Name.
- [x] Introduces divider in CU task view with 1px padding and color aligning to figma design.

### Testing Criteria

image

### CODE REVIEW

File: src/components/cards/TaskCard.tsx [hunk 1]
```diff
@@ -37,8 +39,11 @@ export const TaskCard = ({ task }: { task: TaskResponse }) => {
           <Typography variant="sm" fontSize="12px" sx={{ color: (theme) => theme.color.gray[500] }}>
             {(currentAssignee as IAssigneeCombined).name === 'No assignee'
               ? 'No assignee'
-              : (currentAssignee as IAssigneeCombined)?.name ||
-                `${(currentAssignee as IAssigneeCombined)?.givenName ?? ''} ${(currentAssignee as IAssigneeCombined)?.familyName ?? ''}`.trim()}
+              : truncateText(
+                  (currentAssignee as IAssigneeCombined)?.name ||
+                    `${(currentAssignee as IAssigneeCombined)?.givenName ?? ''} ${(currentAssignee as IAssigneeCombined)?.familyName ?? ''}`.trim(),
+                  TruncateMaxNumber.TASK_CARD_NAME,
+                )}
```
Comment (aatbip) [inline src/components/cards/TaskCard.tsx:46]: I think this computation is unnecessary overload to the app. I think it is better to create an `hoc` that supports trucating with the styling below. We can achieve this using simple css styling - 
```css
width: 146px; //the size of visible element
overflow: hidden;
text-overflow: ellipsis;
white-space: nowrap;
```

File: src/components/cards/ClientTaskCard.tsx [hunk 2]
```diff
@@ -35,89 +35,92 @@ export const ClientTaskCard = ({
...
+                      whiteSpace: 'nowrap',
+                      overflow: 'hidden',
+                    }}
+                  >
+                    {truncateText(
+                      (currentAssignee as IAssigneeCombined)?.name ||
+                        `${(currentAssignee as IAssigneeCombined)?.givenName ?? ''} ${(currentAssignee as IAssigneeCombined)?.familyName ?? ''}`.trim() ||
+                        'No Assignee',
+                      TruncateMaxNumber.SELECTOR,
+                    )}
```
Comment (aatbip) [inline src/components/cards/ClientTaskCard.tsx:100]: I think we are doing two things for the same purpose here. We are giving the above styling in `sx` prop which is exactly responsible for the truncate styling. We don't need to use `truncateText` function in this case.

File: src/types/constants.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ export enum TruncateMaxNumber {
   ATTACHMENT = 15,
   SELECTOR = 16,
   CLIENT_TASK_DESCRIPTION = 256,
+  TASK_CARD_NAME = 22,
```
Comment (aatbip) [inline src/types/constants.ts:5]: I think all of these can be removed and simply use the css styling for truncation. We can create an `hoc` for re-usability. 

I think using such numbers for truncation will bring inconsistency and won't be usable in all kinds of design. It's better to let CSS handle such things for various screen sizes. Styling related jobs should be given to CSS as much as possible for easy life.

**Review by aatbip — CHANGES_REQUESTED** [review]

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Review by rrojan — APPROVED** [review]
LGTM again

**Review by rrojan — APPROVED** [review]

**Review by aatbip — APPROVED** [review]

**Pull request merged**